src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,77 @@
+import { KeyboardEvent, ReactElement, useRef } from "react"
+import { TabList } from "./components/TabList"
+import { TabProps } from "./components/Tab"
+import { TabPanel } from "./components/TabPanel"
+import { Tab } from "./types"
+
+interface TabsProps {
+  /**
+   * We constrain `children` to be element(s) with a label and children
+   * props (e.g. `Tab`). This is only structural, however.
+   **/
+  children: ReactElement<TabProps> | Array<ReactElement<TabProps>>
+  currentTab: number
+  onTab: (index: number) => void
+}
+
+export function Tabs({ children, onTab, currentTab }: TabsProps) {
+  // We a list of refs for tablist nodes so we can focus them programmatically
```

Comment:
@roblafeve _nitpick alert_ 
```suggestion
  // We use a list of refs for tablist nodes so we can focus them programmatically
```

Reply:
@albatrocity this was very carefully chosen wording and it won't be fixed